app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -216,15 +223,24 @@ class NewMessageViewModel @Inject constructor(
     //region Check mailbox existence
     private val exitSignal: CompletableJob = Job()
 
-    val currentMailbox = viewModelScope.suspendLazy {
-        val mailbox = mailboxController.getMailbox(AccountUtils.currentUserId, AccountUtils.currentMailboxId)
+    private val mailboxRefFlow = MutableSharedFlow<MailboxRef>(
+        replay = 1,
+        onBufferOverflow = BufferOverflow.DROP_OLDEST
+    )
 
-        if (mailbox == null) {
-            exitSignal.complete()
-            awaitCancellation()
+    val currentMailbox: Flow<Mailbox> = mailboxRefFlow
```

Comment:
Put this in private and name it `_currentMailboxFlow`
Then create a `currentMailbox` suspend function that does the `.first()`